src/Kiota.Builder/Kiota.Builder.csproj

```diff
@@ -36,6 +36,7 @@
     <PackageReference Include="Microsoft.OpenApi.Readers" Version="1.4.5" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All" />
     <PackageReference Include="YamlDotNet" Version="12.3.1" />
+    <ProjectReference Include="..\Kiota.Generated\KiotaGenerated.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
```

Comment:
I'm not sure how this is going to play with nuget pack. Can you expand on why it's a separate project?

Reply:
All of the examples I found are using a separate project, also the [reference you linked](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators).

In general Compile Time generation/Macros etc. etc always require a separate compilation unit, so that dependencies of the generator are not mixed with dependencies of the project itself etc.

Reply:
great thanks for sharing that! I'm hoping that nuget is smart enough to understand that OutputItemType="Analyzer" ReferenceOutputAssembly="false" mean it should not be referenced in the nuspec. I'm going to test a few things to confirm that first so we don't get any bad surprise at the time of releasing.

Reply:
Amazing, thanks for the double-check!